pkg/controller/gitopsconfig/controller.go

```diff
@@ -414,24 +414,23 @@ func (r *Reconciler) manageDeletion(instance *gitopsv1alpha1.GitOpsConfig) (reco
 		//we return because we need to wait for the job to stop
 		return reconcile.Result{
 			Requeue:      true,
-			RequeueAfter: time.Minute,
+			RequeueAfter: time.Second * 12,
```

Comment:
**Why decrease?**
Before, when there's been a successful hello-world application deployment and then one decided to delete the gitopsconfig, one had to wait for one minute for the finalizer to be removed. Only then k8s garbage collector could remove the gitopsconfig.

Before:
<details><summary><code>time kubectl delete gitopsconfigs.eunomia.kohls.io --all -n eunomia-hello-world-yaml-demo</code></summary><br><pre>
$ time kubectl delete gitopsconfigs.eunomia.kohls.io --all -n eunomia-hello-world-yaml-demo 
gitopsconfig.eunomia.kohls.io "hello-world-yaml" deleted

real    1m0.731s
user    0m0.131s
sys     0m0.183s
</pre></details>

Now:
<details><summary><code>time kubectl delete gitopsconfigs.eunomia.kohls.io --all -n eunomia-hello-world-yaml-demo</code></summary><br><pre>
$ time kubectl delete gitopsconfigs.eunomia.kohls.io --all -n eunomia-hello-world-yaml-demo 
gitopsconfig.eunomia.kohls.io "hello-world-yaml" deleted

real    0m12.543s
user    0m0.112s
sys     0m0.121s
</pre></details>

**But why 12s exactly?**
Because the job becomes successful [no sooner than 10s](https://github.com/KohlsTechnology/eunomia/blob/master/pkg/controller/gitopsconfig/jobmonitor.go#L187) after being launched. I just had to provide a bit higher value here.
I could've provided 5s (for example) but the next reconciliation loop would see the job pending and again a reconciliation loop would be scheduled (this time in half a minute) so the whole deletion process would take 35s.

Reply:
I don't understand the "no sooner than 10s" explanation. The line you link to in jobmonitor.go sets the Status field of GitOpsConfig. Whereas, AFAIU, manageDeletion checks Status field of a Job object. In what way does the jobmonitor.go link explain that the job becomes successful no sooner than 10s?

To be clear: personally, I'm all for shortening this time span (I find the 1m super annoying).

Reply:
Right, I had some misconception about it - I set it to a fixed value of 5 seconds.